### PR TITLE
Rule definition

### LINKED ISSUE #29: Create interface RuleDefinition
- name: string
- description: string
- resourceGraph  
    - create interface ResourceGraphQueryDefintion

### PR DESCRIPTION
**Pull request by Joshua-Phelps**
closes #29 

let's make sure this is the file structure we want

**Proposed structure**

For interfaces and types:
src/models/types.ts 

For classes:

src/models/commands/scanner.ts

src/models/rules/resource_graph_rule.ts 
src/models/rules/terraform_rule.ts

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -821,7 +821,8 @@
     "@types/chai": {
       "version": "4.2.14",
       "resolved": "https://registry.npmjs.org/@types/chai/-/chai-4.2.14.tgz",
-      "integrity": "sha512-G+ITQPXkwTrslfG5L/BksmbLUA0M1iybEsmCWPqzSxsRRhJZimBKJkoMi8fr/CPygPTj4zO5pJH7I2/cm9M7SQ=="
+      "integrity": "sha512-G+ITQPXkwTrslfG5L/BksmbLUA0M1iybEsmCWPqzSxsRRhJZimBKJkoMi8fr/CPygPTj4zO5pJH7I2/cm9M7SQ==",
+      "dev": true
```
Comment (noelbundick) [inline package-lock.json:825]: I see package-lock.json changes but no updates to package.json. Is this file accidentally included?

File: src/models/types.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface RuleDefinition {
```
Comment (noelbundick) [inline src/models/types.ts:1]: nit: Do we have additional conflicts on 'Rule'? Can we drop 'Definition' on these?

File: src/models/types.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  name: string;
+  description: string;
+  resourceGraph?: ResourceGraphQueryDefinition;
```
Comment (noelbundick) [inline src/models/types.ts:4]: This works great for now. When you add tests, can we also add a DummyRule type that can always return true so we know what the system kind of looks like w/ multiple rule types?

Might be worth reading the TS docs on discriminated unions if you haven't yet: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html?ref=hackernoon.com#discriminating-unions

**Review by noelbundick — CHANGES_REQUESTED** [review]
Please confirm package-lock.json, and we can merge the rest

re: file structure, I'd prefer to keep it minimal unless we have a need for a models folder. src/rules & src/commands seems appropriate? And the more common convention is camelCase for filenames in JS/TS versus snake_case (not that it affects functionality)

**Review by noelbundick — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (Joshua-Phelps) [reply package-lock.json:825]: This is from running npm install after deleting my node modules file. I didn't make any other changes

File: src/models/types.ts (hunk 2 above)
Comment (Joshua-Phelps) [reply src/models/types.ts:1]: We can drop the definition part of the name. I'll do the same for the ResourceGraphQuery as well.

File: src/models/types.ts (hunk 3 above)
Comment (Joshua-Phelps) [reply src/models/types.ts:4]: Adding a DummyRule type sounds like a good idea. Thanks for the link

**Review by noelbundick — APPROVED** [review]

**Pull request merged**